stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/attachments/media/MediaAttachmentFragment.kt

```diff
@@ -0,0 +1,146 @@
+import com.getstream.sdk.chat.utils.PermissionChecker
+import com.getstream.sdk.chat.utils.StorageHelper
+import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.ui.attachments.AttachmentSelectionListener
+import io.getstream.chat.android.ui.databinding.StreamFragmentAttachmentMediaBinding
+import io.getstream.chat.core.internal.coroutines.DispatcherProvider
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+public class MediaAttachmentFragment : Fragment() {
+
+    private var _binding: StreamFragmentAttachmentMediaBinding? = null
+    private val binding get() = _binding!!
+
+    private val storageHelper: StorageHelper = StorageHelper()
+    private val permissionChecker: PermissionChecker = PermissionChecker()
+
+    private val gridLayoutManager = GridLayoutManager(context, SPAN_COUNT, RecyclerView.VERTICAL, false)
+    private val gridSpacingItemDecoration = GridSpacingItemDecoration(SPAN_COUNT, SPACING, false)
+    private val mediaAttachmentsAdapter: MediaAttachmentAdapter = MediaAttachmentAdapter {
+        updateMediaAttachment(it)
+    }
+
+    private var selectedAttachments: Set<AttachmentMetaData> = emptySet()
+    private var attachmentSelectionListener: AttachmentSelectionListener? = null
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        attachmentSelectionListener = parentFragment as? AttachmentSelectionListener
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        _binding = StreamFragmentAttachmentMediaBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setupViews()
+        tryPopulateAttachments()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        attachmentSelectionListener?.onAttachmentsSelected(selectedAttachments)
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+
+    override fun onDetach() {
+        super.onDetach()
+        attachmentSelectionListener = null
+    }
+
+    private fun setupViews() {
+        binding.mediaFileRecyclerView.apply {
+            layoutManager = gridLayoutManager
+            addItemDecoration(gridSpacingItemDecoration)
+            adapter = mediaAttachmentsAdapter
+        }
+        binding.grantPermissionsInclude.apply {
+            grantPermissionsImageView.setImageResource(R.drawable.stream_attachment_permission_media)
+            grantPermissionsTextView.setText(R.string.stream_attachment_dialog_permission_media)
+            grantPermissionsTextView.setOnClickListener {
+                tryPopulateAttachments()
+            }
+        }
+    }
+
+    private fun tryPopulateAttachments() {
```

Comment:
Seems it should be called `checkPermissions`

Reply:
Changed corresponding method names in `CameraAttachmentFragment`, `FileAttachmentFragment`, `MediaAttachmentFragment`